thrust/thrust/system/hpx/detail/reduce.h

```diff
@@ -25,6 +29,37 @@
 #elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
 #  pragma system_header
 #endif // no system header
+#include <thrust/detail/function.h>
+#include <thrust/system/hpx/detail/execution_policy.h>
+
+#include <hpx/parallel/algorithms/reduce.hpp>
+
+THRUST_NAMESPACE_BEGIN
+namespace system
+{
+namespace hpx
+{
+namespace detail
+{
+
+template <typename DerivedPolicy, typename InputIterator, typename OutputType, typename BinaryFunction>
+OutputType reduce(
+  execution_policy<DerivedPolicy>&, InputIterator first, InputIterator last, OutputType init, BinaryFunction binary_op)
+{
+  // wrap binary_op
+  thrust::detail::wrapped_function<BinaryFunction, OutputType> wrapped_binary_op{binary_op};
+
+  if constexpr (::hpx::traits::is_forward_iterator_v<InputIterator>)
+  {
+    return ::hpx::reduce(::hpx::execution::par, first, last, init, wrapped_binary_op);
```

Comment:
Shouldn't the first argument be passed along as an HPX execution policy? I thought we could use e.g., `hpx::execution::par` when invoking `thrust::reduce`. But I'm not sure how the dispatching inside CCCL works...

Reply:
Eventually yes, or something like it.. right now the policy is only used internally by thrust's dispatcher, or explicitly by the user in the form `thrust::hpx::par`